### PR TITLE
Increase the test coverage even more

### PR DESCRIPTION
**Pull request by BvB93**
See https://github.com/SCM-NV/qmflows/issues/169.

Major
------
* Added extensive tests for ``qmflows.utils``.
* Added extensive tests for ``qmflows.backports``.
* Added extensive tests for ``qmflows.warnings_qmflows``.
* Added tests for PEP8 compliance.

Minor
------
* Cleaned up type hints in ``qmflows.utils``.
* Rearranged ``qmflows.backports`` to make it more easily testable.
* Removed ``"nbsphinx"`` from ``extras_require["test"]`` in ``setup.py``.
* Added ``"typing_extensions"`` and ``"pycodestyle"``  to ``extras_require["test"]`` in ``setup.py``.
* Removed all remaining references to ``settings2Dict()`` and ``dict2Settings()``.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -49,7 +49,8 @@ def readme():
                       'plams@git+https://github.com/SCM-NV/PLAMS@master',
                       'pyparsing', 'pyyaml>=5.1', 'filelock'],
     extras_require={
-        'test': ['pytest', 'pytest-cov', 'pytest-mock', 'nbsphinx', 'assertionlib'],
+        'test': ['pytest', 'pytest-cov', 'pytest-mock', 'typing_extensions',
```
Comment (felipeZ) [inline setup.py]: `typing_extensions` is good but we should try to **nudge** our user to go for more modern Python version ;)

File: src/qmflows/utils.py [hunk 2]
```diff
@@ -20,30 +19,6 @@
 from .type_hints import PathLike
 
 
-def settings2Dict(s: Settings) -> dict:
```
Comment (felipeZ) [inline src/qmflows/utils.py]: Thank you for removing this!!

File: test/test_backports.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import sys
+from typing import Type, Union
+
+from assertionlib import assertion
+
+from qmflows.backports import (Literal, Final, nullcontext,
+                               _LiteralBackup, _FinalBackup, _NullContextBackup)
+
+
+def test_literal():
```
Comment (felipeZ) [inline test/test_backports.py]: top!

File: test/test_backports.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    """Tests for :data:`Literal` and :data:`_LiteralBackup`."""
+    if sys.version_info.minor >= 8:
+        assertion.eq(Literal.__module__, 'typing')
+    else:
+        assertion.eq(Literal.__module__, 'typing_extensions')
+
+    literal = _LiteralBackup()
```
Comment (felipeZ) [inline test/test_backports.py]: Could you explain to me what `_LiteralBackup` doing?

File: test/test_lint.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+"""Test for PEP8 compliance."""
```
Comment (felipeZ) [inline test/test_lint.py]: Thanks for introducing this very necessary tests!

File: test/test_using_plams/test_failure_package.py [hunk 6]
```diff
@@ -1,31 +1,38 @@
...
-                    job_name="failed_DFTB")
-    fail_adf = adf(None, opt_dftb.molecule, job_name="fail_adf")
-    result = run(fail_adf.molecule, path=tmpdir)
-    print(result)
-    assertion.eq(result, None)
+    # Temporary mute all QMFlows_Warnings
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=QMFlows_Warning)
+
+        # 5 membered ring from which ozone will dissociate
```
Comment (felipeZ) [inline test/test_using_plams/test_failure_package.py]: Could you please delete this legacy (stone age) comment ;)

File: test/test_using_plams/test_failure_package.py [hunk 7]
```diff
@@ -1,31 +1,38 @@
...
-
-    # Calculate the DFTB hessian
-    opt_dftb = dftb(templates.geometry.overlay(dftb_set), mol,
-                    job_name="failed_DFTB")
-    fail_adf = adf(None, opt_dftb.molecule, job_name="fail_adf")
-    result = run(fail_adf.molecule, path=tmpdir)
-    print(result)
-    assertion.eq(result, None)
+    # Temporary mute all QMFlows_Warnings
+    with warnings.catch_warnings():
```
Comment (felipeZ) [inline test/test_using_plams/test_failure_package.py]: Great!

File: test/test_backports.py (hunk 4 above)
Comment (BvB93) [reply test/test_backports.py]: It's creates an instance whose ``__getitem__()`` method has the same signature as ``typing.Literal`` (which was only added in Python 3.8).
As properly backporting ``typing.Literal`` is kinda difficult, it just converts the passed name into the appropriate object type.

```
from typing import Union, Type

Literal = _LiteralBackup()

Literal[1] == int
Literal[int] == Type[int]
Literal[1, 'two', 3.0] == Union[int, str, float]
```

File: setup.py (hunk 1 above)
Comment (BvB93) [reply setup.py]: True, but I feel this subject should be moved to an (eventual) discusion about dropping older python version.
Besides, some of the stuff in ``typing_extensions`` was only added to ``typing`` in 3.8.

File: test/test_using_plams/test_failure_package.py (hunk 6 above)
Comment (BvB93) [reply test/test_using_plams/test_failure_package.py]: And done.

File: setup.py (hunk 1 above)
Comment (felipeZ) [reply setup.py]: Agree

**Pull request merged**